src/app-layer-htp-mem.c

```diff
@@ -58,9 +62,18 @@ void HTPParseMemcap()
                        conf_val);
             exit(EXIT_FAILURE);
         } else {
-            SC_ATOMIC_SET(htp_config_memcap, memcap);
+            if (GlobalMemcapReached(memcap)) {
```

Comment:
same here: this looks a lot like stream, ftp and maybe other places. A util would be much nicer.